From 0046223910f9a63a034474b42ef8b0bd67ba9585 Mon Sep 17 00:00:00 2001
From: Adam Stolcenburg <adam_stolcenburg@comcast.com>
Date: Tue, 6 Oct 2026 17:58:15 +0000
Subject: [PATCH 04/12] Set TZ from Firebolt Localization.timeZone

When the TZ environment variable is not defined, read Localization.timeZone,
and set TZ to it.
---
 .../rdk/shared/platform/firebolt_interface.cc | 22 +++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/src/third_party/starboard/rdk/shared/platform/firebolt_interface.cc b/src/third_party/starboard/rdk/shared/platform/firebolt_interface.cc
index 2a06004..a97bc8b 100644
--- a/src/third_party/starboard/rdk/shared/platform/firebolt_interface.cc
+++ b/src/third_party/starboard/rdk/shared/platform/firebolt_interface.cc
@@ -29,6 +29,8 @@
 
 #include <chrono>
 #include <cstring>
+#include <cstdlib>
+#include <ctime>
 
 using namespace std::chrono_literals;
 using ::starboard::MimeSupportabilityCache;
@@ -66,6 +68,25 @@ std::ostream& operator<<(std::ostream& out, const Firebolt::Error& e) {
   return out << to_string(e) << '(' << static_cast<int32_t>(e) << ')';
 }
 
+void SetTimeZoneFromFirebolt() {
+  if (const char* tz = getenv("TZ"); tz != nullptr) {
+    SB_LOG(INFO) << "TZ is already set to: '" << tz << "'";
+    return;
+  }
+
+  auto time_zone = Firebolt::IFireboltAccessor::Instance().LocalizationInterface().timeZone();
+  if (!time_zone) {
+    SB_LOG(ERROR) << "localization.timeZone() failed, error code = " << time_zone.error();
+    return;
+  }
+  if (time_zone->empty()) {
+    return;
+  }
+  SB_LOG(INFO) << "localization.timeZone(): " << *time_zone;
+  setenv("TZ", (":" + *time_zone).c_str(), 1);
+  tzset();
+}
+
 } // namespace
 
 // FireboltDevice
@@ -596,6 +617,7 @@ void FireboltInterface::lazy_init() {
     const auto connected_tp = std::chrono::steady_clock::now();
     device_.init();
     text_to_speech_.init();
+    SetTimeZoneFromFirebolt();
     if (cfg.legacyRPCv1) {
       // Let gateway know we're ready to receive notifications
       auto& gateway = Firebolt::Transport::GetGatewayInstance();
-- 
2.25.1

